CMakeLists.txt

```diff
@@ -31,6 +31,17 @@ include(AwsSharedLibSetup)
 include(AwsSanitizers)
 include(CheckCCompilerFlag)
 include(AwsFindPackage)
+include(CheckIncludeFile)
+
+# Platform specific header checks
+# Check for if_link.h - needed for REL5 support as REL5 does not have if_link.h
+CHECK_INCLUDE_FILE("linux/if_link.h" LINUX_HAVE_IF_LINK_H)
```

Comment:
I would rather see this be a feature test that follows the same pattern we use in aws-c-common.  See AWS_HAVE_EXECINFO in aws-c-common for example.